src/pose-sensor.js

```diff
@@ -0,0 +1,170 @@
+const SENSOR_FREQUENCY = 16.6;
+
+const X_AXIS = new Vector3(1, 0, 0);
+const Z_AXIS = new Vector3(0, 0, 1);
+
+let orientation = {};
+if (screen.orientation) {
+  orientation = screen.orientation;
+} else if (screen.msOrientation) {
+  orientation = screen.msOrientation;
+} else {
+  Object.defineProperty(orientation, "angle", {
+    get: () => { return (window.orientation || 0) }
+  });
+}
+
+// Quaternion to rotate from sensor coordinates to WebVR coordinates
+const SENSOR_TO_VR = new Quaternion();
+SENSOR_TO_VR.setFromAxisAngle(X_AXIS, -Math.PI / 2);
+SENSOR_TO_VR.multiply(new Quaternion().setFromAxisAngle(Z_AXIS, Math.PI / 2));
+
+/**
+ * An abstraction class around either using the new RelativeOrientationSensor,
+ * or `devicemotion` events with complimentary filter via fusion-pose-sensor.js.
+ */
+export default class PoseSensor {
+  constructor(config) {
+    this.config = config;
+    this.sensor = null;
+    this.fusionSensor = null;
+    this._out = new Float32Array(4);
+
+    // Can be 'sensor' (using RelativeOrientationSensor) or
+    // 'devicemotion' (using devicemotion events via FusionPoseSensor)
+    this.api = 'sensor';
+
+    // Store any errors from Sensors
+    this.error = null;
+
+    // Quaternions for caching transforms
+    this._sensorQ = new Quaternion();
+    this._worldToScreenQ = new Quaternion();
+    this._outQ = new Quaternion();
+
+    this._onSensorRead = this._onSensorRead.bind(this);
+    this._onSensorError = this._onSensorError.bind(this);
+    this._onOrientationChange = this._onOrientationChange.bind(this);
+
+    this._onOrientationChange();
+    this.init();
+  }
+
+  init() {
+    // Attempt to use the RelativeOrientationSensor from Generic Sensor APIs.
+    // First available in Chrome M63, this can fail for several reasons, and attempt
+    // to fallback to devicemotion. Failure scenarios include:
+    //
+    // * Generic Sensor APIs do not exist, fallback to devicemotion
+    // * Underlying sensor does not exist, no fallback possible.
+    // * Feature policy failure (in an iframe); no fallback.
+    // * Permission to sensor data denied; respect user agent, no fallback to devicemotion.
+    //   Browsers are heading towards disabling devicemotion when sensors are denied as well.
+    let sensor = null;
+    try {
+      sensor = new RelativeOrientationSensor({ frequency: SENSOR_FREQUENCY });
+      sensor.addEventListener('error', this._onSensorError);
+    } catch (error) {
+      this.error = error;
+
+      if (error.name === 'SecurityError') {
+        console.error('Cannot construct sensors due to the feature policy');
+      } else if (error.name === 'ReferenceError') {
+        // Fallback to devicemotion
+        this.api = 'devicemotion';
+        this.useDeviceMotion();
+      }
+    }
+
+    if (sensor) {
+      this.sensor = sensor;
+      this.sensor.addEventListener('reading', this._onSensorRead);
+      this.sensor.start();
```

Comment:
perhaps you addressed this earlier and I missed it, but I think you need to [`try`/`catch`](https://w3c.github.io/sensors/#sensor-start) this method call. it could throw `NotReadableError` or `NotReadableError` or `NotAllowedError`, per the [spec](https://w3c.github.io/sensors/#sensor-start).